packages/sidecar/src/main.ts

```diff
@@ -132,12 +152,20 @@ function streamRequestHandler(buffer: MessageBuffer<Payload>, incomingPayload?:
       });
 
       stream.on('end', () => {
-        buffer.put([`${req.headers['content-type']}`, body]);
+        let contentType = req.headers['content-type']?.split(';')[0].toLocaleLowerCase();
+        if (searchParams?.get('sentry_client')?.startsWith('sentry.javascript.browser') && req.headers.origin) {
+          // This is a correction we make as Sentry Browser SDK may send messages with text/plain to avoid CORS issues
+          contentType = 'application/x-sentry-envelope';
```

Comment:
out of curiousity: What kind of problem did this previously cause?

Reply:
I've been trying to use the standalone mode for end to end tracing in a Rails app. Since Spotlight was not on the same page, the Sentry integration did not work properly. I then decided to use the hacked DSN approach to force enable the browser SDK which sent the events with `text/plain;charset=UTF-8` content type. Sidecar simply passed this down to the overlay. The Sentry integration there only listens to the envelope content type so doesn't even get triggered. I tried adding text/plain but that didn't work either way we didn't strip the charset part or do case normalization. It also felt wrong to hook onto text/plain. Hence this solution 😅

Reply:
Ahh this makes a lot of sense. Reminds me that we should export the sentry integration from the `@spotlight/spotlight`(?) package so that users can register it in their browser SDK setup. https://github.com/getsentry/spotlight/issues/403

Reply:
Was gonna look into that but I also noticed there's already a Spotlight integration in the node SDK: https://github.com/getsentry/sentry-javascript/blob/develop/packages/node/src/integrations/spotlight.ts#L6

Maybe just move that into common, enable when `spotlight: true` is passed to init and remove the one from Spotlight? I feel like that's the better approach. WDYT?

Reply:
We can't do this in the browser because this means the spotlight integration would be part of the bundle. Bundlers can't tree shake the integration if we control adding it with a runtime flag like `spotlight: true`.

So for browser SDKs, users will always need to register the integration in `Sentry.init` (like `browserTracingIntegration` today). 

Also I don't think we can easily hoist the same integration into `core` because the sending mechanism (node `http` vs `fetch`) will differ.

Reply:
RE exporting it from `@sentry/browser` vs `@spotlightjs/spotlight` I don't particularly care. We can also export it from the SDKs, as long as we keep things opt-in and tree-shakeable.